packages/mui-material/src/FormHelperText/FormHelperText.d.ts

```diff
@@ -54,6 +54,7 @@ export interface FormHelperTextOwnProps {
     'standard' | 'outlined' | 'filled',
     FormHelperTextPropsVariantOverrides
   >;
+  [key: `data-${string}`]: unknown;
```

Comment:
We shouldn't do this. Look at the solution shown in this comment: https://github.com/mui/material-ui/issues/47230#issuecomment-3514971608. We need to add a type `TextFieldFormHelperTextSlotPropsOverrides` which the developer can extend on their side.